### ISSUE TITLE
Providing project tokens to //api/v1/taxon_names/autocomplete causes failure

### ISSUE THREAD

**Issue by gdower** [SpeciesFileGroup/taxonworks#1620]
**Describe the bug**

When providing a project token to the /taxon_names/autocomplete API endpoint, the request fails:

GET /api/v1/taxon_names/autocomplete?project_id=14&term=buck&token=XXXXXX&project_token=YYYYYY HTTP/1.1
> Host: sandpaper.taxonworks.org
> User-Agent: insomnia/2020.2.2
> Authorization: Bearer XXXXXX
> Accept: */*
```JSON
{
  "success": false
}
```
The same request without a user token and project token works:

GET /api/v1/taxon_names/autocomplete?project_id=14&term=buck HTTP/1.1
> Host: sandpaper.taxonworks.org
> User-Agent: insomnia/2020.2.2
> Authorization: Bearer XXXXXX
> Accept: */*
```JSON
[
  {
    "id": 175265,
    "label": "Cyphoderris buckelli",
    "label_html": "<span class=\"klass\"><i>Cyphoderris <mark>buck</mark>elli</i></span>&nbsp;<span class=\"feedback feedback-info feedback-thin\">species</span>&nbsp;<span class=\"feedback feedback-secondary feedback-thin\"><i>Cyphoderris</i></span>"
  }
]
```

If you DM me on gitter, I can provide the tokens I used, although I know they are valid from working with other API endpoints:

> GET /api/v1/both_authenticated?token=XXXXXX&project_token=YYYYYY HTTP/1.1
> Host: sandpaper.taxonworks.org
> User-Agent: insomnia/2020.2.2
> Authorization: Bearer XXXXXX
> Accept: */*
```JSON
{
  "success": true,
  "open_projects": [
    {
      "XXXXXX": "Testing"
    }
  ]
}
````

**Expected behavior**

Although it is an open project, providing extra authentication credentials probably should not result in the request failing.

**Environment (please identify where you experience this bug:**
- sandpaper.taxonworks.org

**Comment by LocoDelAssembly** [comment]
Does https://sandpaper.taxonworks.org/api/v1/taxon_names/autocomplete?term=buck&project_token=LVEBXJwaRnqSxHmBuRpLQQ work?

Grabbed the token from https://sandpaper.taxonworks.org/api/v1/

**Comment by gdower** [comment]
Yes, although it behaves differently in a browser. I sent you an example in Gitter with my user token where it does fail.

**Comment by mjy** [comment]
@gdower Don't worry about sharing sandbox tokens "in public".  Sooner or later we're going to have people pickup tokens and do something silly with them that will mean we will have to act.  It's better to discover that to the degree we can in sandboxes than in production.

**Comment by gdower** [comment]
Okay, well here's what I meant then:

https://sandpaper.taxonworks.org/api/v1/taxon_names/autocomplete?term=buck&project_token=LVEBXJwaRnqSxHmBuRpLQQ
https://sandpaper.taxonworks.org/api/v1/taxon_names/autocomplete?project_id=14&term=buck&token=r5sYcf-jE2cPFQRKc-ytpA&project_token=LVEBXJwaRnqSxHmBuRpLQQ

**Comment by mjy** [comment]
does the project_id correspond to the project token?

**Comment by LocoDelAssembly** [comment]
It does, but IMHO this is expected behavior, you either use user-token+project_id or project-token to select a project.

Thought this was not documented but actually is! https://api.taxonworks.org/#/autocomplete-get (docs for headers and token and project_id query params)

However reasons are not explained, they are: do not send redundant information and prevent leaking information (probing project_id to detect which one the token belongs to by trying different numbers).

**Comment by gdower** [comment]
Makes sense.

**Comment by mjy** [comment]
@LocoDelAssembly to confirm- you are exactly right, intended behaviour.
